prov/util/src/util_cntr.c

```diff
@@ -182,6 +182,29 @@ static int ofi_cntr_wait(struct fid_cntr *cntr_fid, uint64_t threshold, int time
 	return ret;
 }
 
+int ofi_cntr_spin_wait(struct fid_cntr *cntr_fid, uint64_t threshold,
+		       int timeout)
+{
+	struct util_cntr *cntr;
+	uint64_t endtime, errcnt;
+
+	cntr = container_of(cntr_fid, struct util_cntr, cntr_fid);
+	errcnt = ofi_atomic_get64(&cntr->err);
+	endtime = ofi_timeout_time(timeout);
+
+	do {
+		cntr->progress(cntr);
+		if (ofi_atomic_get64(&cntr->cnt) >= threshold)
+			return FI_SUCCESS;
+
+		if (errcnt != ofi_atomic_get64(&cntr->err))
+			return -FI_EAVAIL;
+
+		if (ofi_adjust_timeout(endtime, &timeout))
+			return -FI_ETIMEDOUT;
+	} while (1);
+}
```

Comment:
Since you exposed the counter calls in order to allow providers to set their own, this call should be adjusted to call fi_cntr_read() and fi_cntr_readerr()